src/python/zquantum/core/wip/circuit/conversions/cirq_conversions.py

```diff
@@ -0,0 +1,235 @@
+
+    def _rotation(angle: Param) -> cirq.EigenGate:
+        return eigengate_cls(
+            global_shift=global_shift, exponent=angle_to_exponent(angle)
+        )
+
+    return _rotation
+
+
+# Mapping between Orquestra gate classes and Cirq gates.
+# Note that not all gates are included, those require special treatment.
+ORQUESTRA_TO_CIRQ_MAPPING = {
+    X: cirq.X,
+    Y: cirq.Y,
+    Z: cirq.Z,
+    RX: cirq.rx,
+    RY: cirq.ry,
+    RZ: cirq.rz,
+    T: cirq.T,
+    I: cirq.I,
+    H: cirq.H,
+    CZ: cirq.CZ,
+    CNOT: cirq.CNOT,
+    SWAP: cirq.SWAP,
+    PHASE: make_rotation_factory(cirq.ZPowGate),
+    CPHASE: make_rotation_factory(cirq.CZPowGate),
+    XX: make_rotation_factory(cirq.XXPowGate, -0.5),
+    YY: make_rotation_factory(cirq.YYPowGate, -0.5),
+    ZZ: make_rotation_factory(cirq.ZZPowGate, -0.5),
+    XY: make_rotation_factory(cirq.ISwapPowGate, 0.0),
+}
+
+
+EIGENGATE_SPECIAL_CASES = {
+    (type(cirq.X), cirq.X.global_shift, cirq.X.exponent): X,
+    (type(cirq.Y), cirq.Y.global_shift, cirq.Y.exponent): Y,
+    (type(cirq.Z), cirq.Z.global_shift, cirq.Z.exponent): Z,
+    (type(cirq.T), cirq.T.global_shift, cirq.T.exponent): T,
+    (type(cirq.H), cirq.H.global_shift, cirq.H.exponent): H,
+    (type(cirq.CNOT), cirq.CNOT.global_shift, cirq.CNOT.exponent): CNOT,
+    (type(cirq.CZ), cirq.CZ.global_shift, cirq.CZ.exponent): CZ,
+    (type(cirq.SWAP), cirq.SWAP.global_shift, cirq.SWAP.exponent): SWAP,
+}
+
+
+EIGENGATE_ROTATIONS = {
+    (cirq.XPowGate, -0.5): RX,
+    (cirq.YPowGate, -0.5): RY,
+    (cirq.ZPowGate, -0.5): RZ,
+    (cirq.ZPowGate, 0): PHASE,
+    (cirq.CZPowGate, 0): CPHASE,
+    (cirq.XXPowGate, -0.5): XX,
+    (cirq.YYPowGate, -0.5): YY,
+    (cirq.ZZPowGate, -0.5): ZZ,
+    (cirq.ISwapPowGate, 0.0): XY,
+}
+
+
+def exponent_to_angle(exponent: Param) -> Param:
+    """Convert exponent from Cirq gate to angle usable in rotation gates..
+
+    Args:
+        exponent: Exponent to be converted.
+    Returns:
+        exponent multiplied by pi.
+    Notes:
+        Scaling of the exponent preserves its "type", i.e. numerical exponents
+        are scaled by numerical approximation of pi, but symbolic ones
+        are scaled by sympy.pi
+    """
+    return exponent * (sympy.pi if isinstance(exponent, sympy.Expr) else np.pi)
+
+
+def angle_to_exponent(angle: Param) -> Param:
+    """Convert exponent from Cirq gate to angle usable in rotation gates..
+
+    Args:
+        angle: Exponent to be converted.
+    Returns:
+        angle divided by pi.
+    Notes:
+        Scaling of the angle preserves its "type", i.e. numerical angles
+        are scaled by numerical approximation of pi, but symbolic ones
+        are scaled by sympy.pi
+    """
+    return angle / (sympy.pi if isinstance(angle, sympy.Expr) else np.pi)
+
+
+@singledispatch
+def convert_to_cirq(obj):
+    """Convert native Orquestra object to its closest Cirq counterpart.
+
+    TODO: add longer description once all conversions are done.
```

Comment:
Should this TODO be addressed before merge or at a later time?

Reply:
@dexter2206 says we should leave it for now